networkx/algorithms/chains.py

```diff
@@ -49,6 +49,11 @@ def chain_decomposition(G, root=None):
     The worst-case running time of this implementation is linear in the
     number of nodes and number of edges [1]_.
 
+    Examples
+    --------
+    >>> G = nx.path_graph(5)
+    >>> chain_d = nx.chain_decomposition(G)
+
```

Comment:
Thanks for adding the example @Lukong123 , but I think we'll need to work on it a bit more to make it really useful for docstring readers!

In it's current state, the example shows how to *call* the function, but doesn't really give the user any sense of how the function behaves. The first challenge here is to come up with a good example graph - you want to choose one that has properties which are relevant for the function in the example (e.g. in this case, the example graph should probably have a chain).

Typically the best way to give users a sense of what a function does is to show it's output. In this case, `chain_decomposition` actually creates a [generator](https://wiki.python.org/moin/Generators) which isn't really that informative until it is iterated over! For example, one easy way to consume the objects from a generator is to create a list from it, e.g. `list(nx.chain_decomposition(G))`. Note how the result is empty (this goes back to the first point of choosing a good example graph)!

Anyways these are just a few things to keep in mind when coming up with good docstring examples!

Reply:
Thank you, sir, I will keep that in mind.